### PR TITLE
fix #1151 fixed some warnings

### LINKED ISSUE #1151: Remove warnings in debug
Due to missing keys or wrong property checks

image

### PR DESCRIPTION
**Pull request by saidaipparla**

### CODE REVIEW

File: web/client/components/maps/MapGrid.jsx [hunk 1]
```diff
@@ -90,7 +90,7 @@ var MapGrid = React.createClass({
     renderMetadataModal() {
         if (this.props.metadataModal) {
             let MetadataModal = this.props.metadataModal;
-            return (<MetadataModal ref="metadataModal" show={this.props.currentMap && this.props.currentMap.displayMetadataEdit} onHide={this.props.resetCurrentMap}
+            return (<MetadataModal key="metadataModal" show={this.props.currentMap && this.props.currentMap.displayMetadataEdit} onHide={this.props.resetCurrentMap}
```
Comment (mbarto) [inline web/client/components/maps/MapGrid.jsx]: You cannot remove the ref, it's used

File: web/client/plugins/CreateNewMap.jsx [hunk 2]
```diff
@@ -17,15 +17,15 @@ const CreateNewMap = React.createClass({
         mapType: React.PropTypes.string,
         onGoToMap: React.PropTypes.func,
         colProps: React.PropTypes.object,
-        isLoggedIn: React.PropTypes.bool
+        isLoggedIn: React.PropTypes.object
```
Comment (mbarto) [inline web/client/plugins/CreateNewMap.jsx]: bool is correct. You need to change the connect to set a bool value

File: web/client/plugins/CreateNewMap.jsx [hunk 3]
```diff
@@ -17,15 +17,15 @@ const CreateNewMap = React.createClass({
...
     },
     contextTypes: {
         router: React.PropTypes.object
     },
     getDefaultProps() {
         return {
             mapType: "leaflet",
-            isLoggedIn: false,
+            isLoggedIn: "",
```
Comment (mbarto) [inline web/client/plugins/CreateNewMap.jsx]: false is correct

File: web/client/components/maps/MapGrid.jsx (hunk 1 above)
Comment (saidaipparla) [inline web/client/components/maps/MapGrid.jsx]: ok i will keep ref and add key

**Comment by saidaipparla** [comment]
yes as far i have seen there are other two warnings  
- when we will click on  print link from menu
- when we will click on settings icon in layer properties

File: web/client/plugins/CreateNewMap.jsx (hunk 3 above)
Comment (saidaipparla) [inline web/client/plugins/CreateNewMap.jsx]: ok i will change this

**Comment by mbarto** [comment]
Ok, then fix also the two remaining warnings

File: web/client/plugins/CreateNewMap.jsx [hunk 4]
```diff
@@ -51,6 +51,6 @@ const CreateNewMap = React.createClass({
 module.exports = {
     CreateNewMapPlugin: connect((state) => ({
         mapType: (state.maps && state.maps.mapType) || (state.home && state.home.mapType),
-        isLoggedIn: state && state.security && state.security.user
+        isLoggedIn: state && state.security && state.security.user && state.security.user.enabled
```
Comment (mbarto) [inline web/client/plugins/CreateNewMap.jsx]: isLoggedIn: state && state.security && state.security.user && true || false

File: web/client/localConfig.json [hunk 5]
```diff
@@ -385,7 +385,7 @@
 					"closeGlyph": "1-close",
 					"submitConfig": {
 						"buttonConfig": {
-							"bsSize": "medium",
+							"bsSize": "small",
```
Comment (mbarto) [inline web/client/localConfig.json]: Have you checked that the button doesn't change?

**Comment by mbarto** [comment]
Is the "when we will click on settings icon in layer properties" still there?

File: web/client/localConfig.json (hunk 5 above)
Comment (saidaipparla) [inline web/client/localConfig.json]: yes i checked its changed

File: web/client/plugins/CreateNewMap.jsx (hunk 4 above)
Comment (saidaipparla) [inline web/client/plugins/CreateNewMap.jsx]: ok i will change

**Pull request merged**